src/panels/config/backup/dialogs/dialog-download-decrypted-backup.ts

```diff
@@ -0,0 +1,228 @@
+import { mdiClose } from "@mdi/js";
+import type { CSSResultGroup } from "lit";
+import { LitElement, css, html, nothing } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import "../../../../components/ha-dialog-header";
+import "../../../../components/ha-icon-button";
+import "../../../../components/ha-icon-next";
+import "../../../../components/ha-md-dialog";
+import type { HaMdDialog } from "../../../../components/ha-md-dialog";
+import "../../../../components/ha-md-list";
+import "../../../../components/ha-md-list-item";
+import "../../../../components/ha-svg-icon";
+import "../../../../components/ha-password-field";
+import "../../../../components/ha-alert";
+import {
+  canDecryptBackupOnDownload,
+  getPreferredAgentForDownload,
+} from "../../../../data/backup";
+import type { HassDialog } from "../../../../dialogs/make-dialog-manager";
+import { haStyle, haStyleDialog } from "../../../../resources/styles";
+import type { HomeAssistant } from "../../../../types";
+import { downloadBackupFile } from "../helper/download_backup";
+import type { DownloadDecryptedBackupDialogParams } from "./show-dialog-download-decrypted-backup";
+
+@customElement("ha-dialog-download-decrypted-backup")
+class DialogDownloadDecryptedBackup extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _opened = false;
+
+  @state() private _params?: DownloadDecryptedBackupDialogParams;
+
+  @query("ha-md-dialog") private _dialog?: HaMdDialog;
+
+  @state() private _encryptionKey = "";
+
+  @state() private _error = "";
+
+  public showDialog(params: DownloadDecryptedBackupDialogParams): void {
+    this._opened = true;
+    this._params = params;
+  }
+
+  public closeDialog() {
+    this._dialog?.close();
+    return true;
+  }
+
+  private _dialogClosed() {
+    if (this._params!.cancel) {
+      this._params!.cancel();
```

Comment:
So we even call cancel when we have submitted succesfully?

Reply:
I copied another dialog. I will remove all this logic as it's not really needed.